### PR TITLE
`Tensor.cumprod`

### PR DESCRIPTION
**Pull request by eitanturok**
Add cumulative product to Tensor using `_cumalu`.

### CODE REVIEW

File: tinygrad/tensor.py [hunk 1]
```diff
@@ -2192,10 +2192,11 @@ def matmul(self, x:Tensor, reverse=False, acc_dtype:Optional[DTypeLike]=None) ->
     return x.dot(self, acc_dtype=acc_dtype) if reverse else self.dot(x, acc_dtype=acc_dtype)
 
   def _cumalu(self, axis:int, op:Ops, _include_initial=False) -> Tensor:
-    assert self.shape[axis] != 0 and op in (Ops.ADD, Ops.MAX)
+    op_to_tensor_op: Dict[Ops, Callable[[Tensor, Optional[int]], Tensor]] = {Ops.ADD: Tensor.sum, Ops.MAX: Tensor.max, Ops.MUL: Tensor.prod}
```
Comment (eitanturok) [inline tinygrad/tensor.py:2195]: cleaner way to write `op_to_tensor_op`?

File: tinygrad/tensor.py [hunk 2]
```diff
@@ -2208,7 +2209,8 @@ def _split_cumalu(self, axis:int, op:Ops) -> Tensor:
     base = ret[..., -1]._cumalu(-1, op, _include_initial=True)
     base = base.unsqueeze(-1).expand(*base.shape, ret.shape[-1])
     def fix(x:Tensor): return x.flatten(start_dim=-2)[..., -s:].transpose(axis,-1)
-    return fix(ret) + fix(base) if op is Ops.ADD else fix(ret).maximum(fix(base))
+    op_to_tensor_op: Dict[Ops, Callable[[Tensor, Tensor], Tensor]] = {Ops.ADD: Tensor.add, Ops.MAX: Tensor.maximum, Ops.MUL: Tensor.mul}
```
Comment (eitanturok) [inline tinygrad/tensor.py:2212]: cleaner way to write `op_to_tensor_op`?

File: tinygrad/tensor.py (hunk 1 above)
Comment (eitanturok) [reply tinygrad/tensor.py:2195]: Maybe use `Tensor.from_uop`. But then we have to use `UOp` instead of `Ops`...

**Comment by chenyuxyz** [comment]
bad tests, please understand what you are writing before writing it

**Comment by eitanturok** [comment]
Got it, bad tests.

I took a closer look at my code and think it was unnecessary to modify the `test/external/external_test_onnx_backend.py` file.

I know tinygrad has a high bar for PRs and was wondering if there are other reasons why these tests are bad. I am trying to learn and would love to make the tests better.

**Pull request closed without merging**